shared/src/main/scala/squants/experimental/unitgroups/si/strict/package.scala

```diff
@@ -13,7 +13,7 @@ package object strict {
   object implicits {
     implicit def mkSiUnitGroup[A <: Quantity[A]](implicit dimension: Dimension[A]): UnitGroup[A] = {
       new UnitGroup[A] {
-        val units: Set[UnitOfMeasure[A]] = dimension.units.collect { case si: SiUnit => si }
+        val units: Set[UnitOfMeasure[A]] = dimension.units.filter(_.isInstanceOf[SiUnit])
```

Comment:
Why is this change required? I'd rather not have `isInstanceOf` in the code base

Reply:
This a bug in Dotty. See lampepfl/dotty#3208. It is not clear if this will be fixed as one can always rewrite the code as:
```scala
dimension.units.collect { case si: UnitOfMeasure[A] with SiUnit => si }
```
However, I though using `collect` here was overkill since what it really doe is `filter`

Reply:
I'd rather have the `collect` call than `isInstanceof`